### PR TITLE
Bug fixes for dual write functionality

### PR DESCRIPTION
**Pull request by jcacciatore**
* Property based settings now work properly
* Fixed bug with injecting HostSupplier
* Now support dual write functionality if client app is using d_ methods in DynoJedisClient

### CODE REVIEW

File: dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoDualWriterClient.java [hunk 1]
```diff
@@ -174,74 +174,106 @@ public void setRange(int range) {
...
+        return this.d_append(key, value).getResult();
+    }
+
+    @Override
+    public OperationResult<Long> d_append(final String key, final String value) {
         writeAsync(key, new Callable<OperationResult<Long>>() {
             @Override
             public OperationResult<Long> call() throws Exception {
-                return d_append(key, value);
+                return DynoDualWriterClient.super.d_append(key, value);
```
Comment (ipapapa) [inline dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoDualWriterClient.java]: Sounds weird that we are calling DynoDualWriterClient instead DynoDualWriterClient class. Wouldn't `this` not work here?

File: dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoDualWriterClient.java [hunk 2]
```diff
@@ -174,74 +174,106 @@ public void setRange(int range) {
 
     @Override
     public Long append(final String key, final String value) {
+        return this.d_append(key, value).getResult();
+    }
+
+    @Override
+    public OperationResult<Long> d_append(final String key, final String value) {
```
Comment (ipapapa) [inline dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoDualWriterClient.java]: was implementing the `d_xxx` needed for the dual write fix, or they were done as part of our conversation?

**Comment by ipapapa** [comment]
Other than my comments, logically LGTM.

File: dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoDualWriterClient.java (hunk 1 above)
Comment (jcacciatore) [inline dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoDualWriterClient.java]: Using `this` will cause an infinite loop. Note that I have to specify `DynoDualWriterClient` in order to call **`super`**

File: dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoDualWriterClient.java (hunk 2 above)
Comment (jcacciatore) [inline dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoDualWriterClient.java]: It was not necessary for the fix. I either needed to support it or modify pappy code calling these methods.

**Comment by ipapapa** [comment]
Merging, tagging and baking to test.

**Pull request merged**